app/src/main/java/net/squanchy/notification/NotificationService.kt

```diff
@@ -0,0 +1,30 @@
+package net.squanchy.notification
+
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import net.squanchy.schedule.domain.view.Event
+import net.squanchy.service.firebase.FirebaseAuthService
+import net.squanchy.service.repository.EventRepository
+import net.squanchy.support.lang.Lists.filter
+import java.util.Collections
+import java.util.Comparator
+
+internal class NotificationService(private val authService: FirebaseAuthService, private val eventRepository: EventRepository) {
+
+    fun sortedFavourites(): Observable<List<Event>> {
+        return authService.ifUserSignedInThenObservableFrom { userId ->
+            eventRepository.events(userId)
+                    .map { events -> filter(events, { it.favorited }) }
```

Comment:
we could use Kotlin filter instead of ours here: `.map { events -> events.filter { it.favorited } }`

Reply:
Good point!

Reply:
Simplifying more:

```kotlin
eventRepository.events(userId)
        .map { it.filter { it.favorited } }
        .map { it.sortedBy { it.startTime } }
        .take(1)
        .subscribeOn(Schedulers.io())
```

Reply:
Oooh even better 👍